### PR TITLE
add externalIPs and kube-proxy test matrix

### PR DESCRIPTION
**Pull request by aanm**
This test matrix is useful for developers to quickly test an old and new
behavior of kube-proxy and eBPF host service implementation.

It will be enabled for all PRs by default.

With this PR, it will check if datapath changes brake the known behavior of Cilium for node-port service with kube-proxy running.

As a follow up, once #9901 is merged we can add the from host test matrix, which will test the behavior for externalIPs services.

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9849)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by brb** [comment]
How long does it take to run the suite? Could we make it as a part of Nightly?

**Comment by aanm** [comment]
> How long does it take to run the suite? Could we make it as a part of Nightly?

@brb the tests themselves no more than 3 minutes I would say. The issue is the test setup.

**Comment by brb** [comment]
@aanm As discussed yesterday during the Cilium dev meeting, we could generate the table with expected outcome and then test Cilium against it. AFAIK @joestringer mentioned about table driven testing in ginkgo for different datapath configurations.

**Comment by joestringer** [comment]
FWIW, we do some table-driven testing here:

https://github.com/cilium/cilium/blob/f543ea2367c7dfc64b3cdc2b871a2a58d5328e58/pkg/policy/distillery_test.go#L417

I previously wrote up the table in this issue with the hope that we can create a table for it eventually: https://github.com/cilium/cilium/issues/8491

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-focus K8sKubeProxyFreeMatrix*

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: test/k8sT/manifests/externalIPs/matrix.bash [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        G ) golang="true";;
+        H ) headers="true";;
+        g ) namespace=${OPTARG};;
+        h ) help="true";;
+
+     esac
+done
+
+if [[ -n "${help}" ]]; then
+        echo "matrih.bash"
```
Comment (nebril) [inline test/k8sT/manifests/externalIPs/matrix.bash]: s/matrih/matrix

**Review by nebril — COMMENTED** [review]
aside from typo in matrix.bash help lgtm

File: test/k8sT/manifests/externalIPs/matrix.bash (hunk 1 above)
Comment (aanm) [reply test/k8sT/manifests/externalIPs/matrix.bash]: fixed typo

**Comment by aanm** [comment]
merging, the PR passed all tests previously, only fixed a typo mentioned by Maciej.

**Comment by brb** [comment]
Too late to the party, but I think it didn't make sense to merge it before #9901 got merged.

**Comment by aanm** [comment]
> Too late to the party, but I think it didn't make sense to merge it before #9901 got merged.

Even if #9901 was not merged the tests introduced in this PR also checks if the behavior of cilium changes with kube-proxy enabled which is also useful to catch potential datapath regressions.

**Pull request merged**